compiler/rustc_passes/src/check_attr.rs

```diff
@@ -147,6 +147,9 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                     | AttributeKind::ConstStabilityIndirect
                     | AttributeKind::MacroTransparency(_),
                 ) => { /* do nothing  */ }
+                Attribute::Parsed(AttributeKind::AsPtr) => {
+                    self.check_applied_to_fn_or_method(hir_id, attr, span, target)
```

Comment:
this is an ICE, `check_applied_to_fn_or_method` runs attr.span() which aren't supported on parsed attributes

Reply:
no it isn't, look 
https://github.com/rust-lang/rust/blob/86039ccfac5ab8884f123b911c870abd3cce938e/compiler/rustc_passes/src/check_attr.rs#L1835

It still calls attr.span()

Reply:
ah right

Reply:
I kind of want to get rid of that function, the few that are in that list are there because it was too painful to migrate otherwise

Reply:
could you solve it properly?